modules/exploits/osx/browser/firefox_audio_events_dom_uaf.rb

```diff
@@ -0,0 +1,183 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+  include Msf::Exploit::Remote::BrowserExploitServer
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => "Firefox audio events RCE",
+      'Description'    => %q{
+        This module exploits a UAF vulnerability present in Firefox's
+        nsNodeUtils::CloneAndAdopt function, which is exploited by adding and
+        immediately removing audio elements. This targets an exact browser
+        version (50.0.X x64 under OSX) and requires an information disclosure
+        leak via CVE-2017-5465. The exploit needs at least 4 GB of ram to operate
+        (the more the better), and can take a few minutes to complete.
+        },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'dcodeIO',             # Stand-alone version for long.js
+          'echo',                # Original vulnrability reporter
```

Comment:
Not sure if intentional.

```suggestion
          'echo',                # Original vulnerability reporter
```